analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParameterValidationInYieldShouldBeWrapped.CSharp10.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Tests.Diagnostics
+{
+    public static class InvalidCases
+    {
+        public static IEnumerable<string> Foo(string something) // FN
+        {
+            ArgumentNullException.ThrowIfNull(something); // FN sec
+
+            yield return something;
+        }
+
+        public static IEnumerable<int> GetSomething(string value) // FN - this is an edge case that might be worth handling later on
+        {
+            yield return 42;
+
+            if (value == null)
+            {
+                ArgumentNullException.ThrowIfNull(value); // FN sec
+            }
```

Comment:
This could be another rule. This just doesn't make sense.

Reply:
I copied the test cases from analyzers\tests\SonarAnalyzer.UnitTest\TestCases\ParameterValidationInYieldShouldBeWrapped.cs and changed `throw new ArgumentNullException(nameof(value));` with `ArgumentNullException.ThrowIfNull(value);`

I wouldn't change the existing test cases for this rule in this PR... this is to document we don't support ArgumentNullException.ThrowIfNull

I could simplify the tests if you believe it's better